src/mlpack/methods/ann/layer/concat_impl.hpp

```diff
@@ -86,62 +69,70 @@ template<typename eT>
 void Concat<InputDataType, OutputDataType, CustomLayers...>::Backward(
     const arma::Mat<eT>&& /* input */, arma::Mat<eT>&& gy, arma::Mat<eT>&& g)
 {
-  size_t outSize = 0;
-  size_t elements = 0;
-
-  for (size_t i = 0, j = 0; i < network.size(); ++i, j += elements)
+  size_t rowCount = 0;
+  if (run)
   {
-    elements = boost::apply_visitor(outputParameterVisitor,
-        network[i]).n_elem;
-
     arma::mat delta;
-    if (gy.n_cols == 1)
-    {
-      delta = gy.submat(j, 0, j + elements - 1, 0);
-    }
-    else
+    for (size_t i = 0; i < network.size(); ++i)
     {
-      delta = gy.submat(0, i, elements - 1, i);
+      size_t rows = boost::apply_visitor(outputParameterVisitor, network[i]).n_rows;
```

Comment:
Do you think we should add a comment here, to provide some insight what this code does?

Reply:
Yes, would be better.